ui/lib/core/addon/components/bar-chart.js

```diff
@@ -0,0 +1,327 @@
+/**
+ * @module BarChart
+ * BarChart components are used to display data in the form of a stacked bar chart, with accompanying legend and tooltip. Anything passed into the block will display in the top right of the chart header.
+ *
+ * @example
+ * ```js
+ * <BarChartComponent @title="Top 10 Namespaces" @description="Each namespace's client count includes clients in child namespaces." @labelKey="namespace_path" @dataset={{this.testData}} @mapLegend={{ array (hash key="non_entity_tokens" label="Active direct tokens") (hash key="distinct_entities" label="Unique Entities") }} @onClick={{this.onClick}} >
+ *    <button type="button" class="link">
+ *      Export all namespace data
+ *    </button>/>
+ * </BarChartComponent>
+ *
+ *  mapLegendSample = [{
+ *     key: "api_key_for_label",
+ *     label: "Label Displayed on Legend"
+ *   }]
+ * ```
+ *
+ * @param {string} title - title of the chart
+ * @param {array} mapLegend - array of objects with key names 'key' and 'label' for the map legend
+ * @param {object} dataset - dataset for the chart
+ * @param {string} [description] - description of the chart
+ * @param {string} [labelKey=label] - labelKey is the key name in the dataset passed in that corresponds to the value labeling the y-axis
+ * @param {function} [onClick] - takes function from parent and passes it to click event on data bars
+ *
+ */
+
+import Component from '@glimmer/component';
+import layout from '../templates/components/bar-chart';
+import { setComponentTemplate } from '@ember/component';
+import { assert } from '@ember/debug';
+import { action } from '@ember/object';
+import { guidFor } from '@ember/object/internals';
+import { scaleLinear, scaleBand } from 'd3-scale';
+import { axisLeft } from 'd3-axis';
+import { max } from 'd3-array';
+import { stack } from 'd3-shape';
+// eslint-disable-next-line no-unused-vars
+import { select, event, selectAll } from 'd3-selection';
+// eslint-disable-next-line no-unused-vars
+import { transition } from 'd3-transition';
+
```

Comment:
Chart styling is within the component file for now so that a future developer doesn't have to hunt down the `.css` file and some `.css` styling is kind of wonky when interacting with d3. In next PR, when cleaning up d3, may consider moving all static styling to `.css`